src/services/enum-utilities.ts

```diff
@@ -137,19 +132,50 @@ export default class EnumUtilities {
     return (item.name === FilingTypes.TRANSITION)
   }
 
-  /** Returns True if filing is an Administrative Dissolution. */
-  static isTypeAdministrativeDissolution (item: any): boolean {
-    return (item.name === FilingTypes.DISSOLUTION && item.dissolutionType === DissolutionTypes.ADMINISTRATIVE)
+  /**
+   * Returns True if filing is an Administrative Dissolution.
+   * @param item a todo item, state filing or filing item
+   */
+  static isTypeDissolutionAdministrative (item: any): boolean {
+    return (
+      item.filingSubType === FilingSubTypes.DISSOLUTION_ADMINISTRATIVE ||
+      item.dissolution?.dissolutionType === FilingSubTypes.DISSOLUTION_ADMINISTRATIVE ||
+      item.data?.dissolution?.dissolutionType === FilingSubTypes.DISSOLUTION_ADMINISTRATIVE
+    )
```

Comment:
I realized this method, and the ones below, are called from different components using different data objects. So, I addd the "param" comment indicating the items the method supports, and the code checks all possibilities.

I don't like this, but fixing it would require some API changes (since the API doesn't provide filing sub-type consistently) and maybe splitting this method into separate cases for API objects or UI objects. I think this is OK for now but I admit it's fragile.

Reply:
Good work @severinbeauvais